src/lib/Timeline.js

```diff
@@ -947,6 +952,32 @@ export default class ReactCalendarTimeline extends Component {
     return label ? <InfoLabel label={label} /> : ''
   }
 
+  renderLeftSidebar = () => {
```

Comment:
Why move these into separate functions? Is this related to optimizations?

Reply:
Yes, it is for optimization reasons since everytime you call the `head` method a new object for `leftSidebar` and `rightSidebar` will get generated and the `Header` is a pure component which will result in extra renders